batch/batch/driver/scheduler.py

```diff
@@ -174,11 +177,13 @@ def allocate_cores(user, mark):
             async for record in records:
                 batch_id = record['batch_id']
                 job_id = record['job_id']
+                attempt_id = ''.join([secrets.choice('abcdefghijklmnopqrstuvwxyz0123456789') for _ in range(6)])
+                record['attempt_id'] = attempt_id
```

Comment:
What's the thinking on moving this here? It seems to me to be semantically equivalent. Is this solving a bug?

Reply:
I think I needed the attempt id before on some previous iteration. I think you're right and it should go on schedule_job.